fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -1142,9 +1146,24 @@ private long loadTxnBeginImpl(TLoadTxnBeginRequest request, String clientIp) thr
             throw new UserException("unknown table \"" + request.getDb() + "." + request.getTbl() + "\"");
         }
 
-        // begin
         long timeoutSecond = request.isSetTimeout() ? request.getTimeout() : Config.stream_load_default_timeout_second;
         MetricRepo.COUNTER_LOAD_ADD.increase(1L);
+
+        // just use default value of session variable
+        // as there is no connectContext for sync stream load
+        ConnectContext connectContext = new ConnectContext();
+        if (connectContext.getSessionVariable().isEnableLoadProfile()) {
+            TransactionResult resp = new TransactionResult();
+            StreamLoadManager streamLoadManager = GlobalStateMgr.getCurrentState().getStreamLoadManager();
+            streamLoadManager.beginLoadTask(dbName, table.getName(), request.getLabel(), timeoutSecond, resp);
+
+            StreamLoadTask task = streamLoadManager.getSyncSteamLoadTaskByLabel(request.getLabel());
+            if (task == null || task.getTxnId() == -1) {
+                throw new UserException("Load begin transactonId failed");
```

Comment:
Add the txnId in the exception to better trace the error.

Reply:
There is no txnId if `begin transaction` failed, just log the label